packages/dashboard/src/components/playground/propertiesTable/propertiesTable.tsx

```diff
@@ -0,0 +1,267 @@
+import Table from '@cloudscape-design/components/table';
+import React from 'react';
+import {
+  Box,
+  Button,
+  CollectionPreferences,
+  Header,
+  SpaceBetween,
+  TextFilter,
+} from '@cloudscape-design/components';
+import Pagination from '@cloudscape-design/components/pagination';
+import './propertiesTable.css';
+import CustomOrangeButton from '~/components/customOrangeButton';
+import { CancelableEventHandler } from '@cloudscape-design/components/internal/events';
+import { ButtonProps } from '@cloudscape-design/components/button/interfaces';
+import { useSelectedWidgets } from '~/hooks/useSelectedWidgets';
+import { AssetSummary } from '~/hooks/useAssetDescriptionQueries';
+import {
+  SiteWiseQueryConfig,
+  StyledAssetQuery,
+} from '~/customization/widgets/types';
+import { useCollection } from '@cloudscape-design/collection-hooks';
+import { useWidgetMetaData } from '~/components/playground/propertiesTable/widgetMetaData/useWidgetMetaData';
+import {
+  spaceScaledL,
+  spaceScaledXl,
+  spaceScaledXs,
+} from '@cloudscape-design/design-tokens';
+
+export const convertPropertiesToMap = (
+  describedAssetsMapQueryData: Record<string, AssetSummary>
+) => {
+  if (Object.entries(describedAssetsMapQueryData).length === 0)
+    return undefined;
+
+  const assetEntries = Object.entries(describedAssetsMapQueryData).map(
+    ([assetId, summary]) => {
+      const propertiesEntries = summary.properties.map((property) => {
+        return [property.propertyId, property];
+      });
+      return [
+        assetId,
+        { ...summary, properties: Object.fromEntries(propertiesEntries) },
+      ];
+    }
+  );
+
+  return Object.fromEntries(assetEntries);
+};
+
+export const PropertiesTable = ({
+  onAddDataStreams,
+}: {
+  onAddDataStreams: CancelableEventHandler<ButtonProps.ClickDetail>;
+}) => {
+  const selectedWidgets = useSelectedWidgets();
+
+  const queryConfig: StyledAssetQuery | undefined = selectedWidgets.length
+    ? (selectedWidgets[0].properties.queryConfig as SiteWiseQueryConfig)?.query
+    : undefined;
+
+  const allItems = useWidgetMetaData(queryConfig);
+
+  const { items, paginationProps, collectionProps } = useCollection(
+    allItems ?? [],
+    {
+      pagination: {
+        defaultPage: 1,
+      },
+      selection: {
+        keepSelection: true,
+        trackBy: 'key',
+      },
+    }
+  );
+  return (
+    <div className='properties-table-parent-container'>
+      <Table
+        stickyHeader
+        onSelectionChange={collectionProps?.onSelectionChange}
+        selectedItems={collectionProps.selectedItems}
+        loadingText='Loading resources'
+        selectionType='multi'
+        trackBy='key'
+        contentDensity='compact'
+        resizableColumns
+        stickyColumns={{ first: 1 }}
+        columnDefinitions={[
```

Comment:
can all the larger props be moved into their own files for legibility

`columnDefinition`
`emptyState`
`header`
`preferences`

Reply:
Yes, this file is 100% WIP, and hence not changing much here. yes this file is the first one to change in the follow up CR

Reply:
I don't understand why not do this now? Why does the WIP state mean we don't improve it now?

Reply:
+1 would push for this to be done sooner rather than later 🙏 
WIP doesn't guarantee you will return to make the changes